copier.yml

```
@@ -0,0 +1,92 @@
+project_name:
+  type: str
+  default: "pybamm-example-project"
+  help: "The name of your project"
+
+project_slug:
+  type: str
+  help: Your Python package name (should be in `snake_case`)
+  default: "{{ project_name|lower|replace('-', '_')|replace(' ', '_') }}"
+  validator: |
+    {% if ('-' in project_slug or ' ' in project_slug) %}
+    Package name should not contain empty spaces or hyphens
+    {% endif %}
+
```

I have some reservations about these fields, since people not acquainted with packaging their code might not understand the difference between what a "package" and a "project" are. In theory, packages, whether uploaded on PyPI or not, can be interchangeably called projects in the greater sense of the programming world, which adds to the confusion. I propose something like this:

```suggestion
project_name:
  type: str
  default: "pybamm-example-project"
  help: "The name of your project"

project_slug:
  type: str
  help: The name of the directory containing the Python files for your project. This should be in `snake_case`, and this is what you will import it with, such as `import pybamm_example_project`.
  default: "{{ project_name|lower|replace('-', '_')|replace(' ', '_') }}"
  validator: |
    {% if ('-' in project_slug or ' ' in project_slug) %}
    The name of the directory should not contain spaces or hyphens
    {% endif %}

```